## Human Datasets

> ### Automatic cell-type harmonization and integration across Human Cell Atlas datasets
> https://cellxgene.cziscience.com/collections/854c0855-23ad-4362-8b77-6b1639e7a9fc
> 
> Harmonizing cell types across the single-cell community and assembling them into a common framework is central to building a standardized Human Cell Atlas. Here, we present CellHint, a predictive clustering tree-based tool to resolve cell-type differences in annotation resolution and technical biases across datasets. CellHint accurately quantifies cell-cell transcriptomic similarities and places cell types into a relationship graph that hierarchically defines shared and unique cell subtypes. Application to multiple immune datasets recapitulates expert-curated annotations. CellHint also reveals underexplored relationships between healthy and diseased lung cell states in eight diseases. Furthermore, we present a workflow for fast cross-dataset integration guided by harmonized cell types and cell hierarchy, which uncovers underappreciated cell types in adult human hippocampus. Finally, we apply CellHint to 12 tissues from 38 datasets, providing a deeply curated cross-tissue database with ∼3.7 million cells and various machine learning models for automatic cell annotation across human tissues.
> 
> Used to get Renal System data for _Homo sapiens_
>
> dataset_h5ad_path: f95d8919-1f2a-405f-8776-bfecc0ab0f3f.h5ad 

> ### An atlas of healthy and injured cell states and niches in the human kidney (Version 1.5)
> https://cellxgene.cziscience.com/collections/0f528c8a-a25c-4840-8fa3-d156fa11086f
> 
> Understanding kidney health and disease relies upon defining the complexity of cell types and states, their associated molecular profiles, and interactions within tissue neighborhoods. We have applied single-cell and single-nucleus assays to a broad spectrum of healthy reference and disease kidneys. This has provided a high-resolution cellular atlas that includes rare and novel cell populations. We further identify and define cellular states altered in kidney injury, encompassing cycling, adaptive or maladaptive repair, transitioning and degenerative states affecting several nephron segments. These analyses further define biological pathways relevant to injury niches, including signatures underlying the transition from reference to predicted maladaptive states that are associated with a decline in kidney function during chronic kidney disease. Our collaborative efforts across the Kidney Precision Medicine Project (KPMP), the Human BioMolecular Atlas Program (HuBMAP) and the Human Cell Atlas consortia aim to generate a comprehensive human kidney cell atlas that includes injury and disease-relevant cell states and neighborhoods as a valuable resource to the research community. Here we provide an expanded data set from that published previously (Lake et al., Nature 2023) that includes additional KPMP biopsy data.
> 
> dataset_h5ad_path: dea717d4-7bc0-4e46-950f-fd7e1cc8df7d
> 
> dataset_h5ad_path: a12ccb9b-4fbe-457d-8590-ac78053259ef

> ### High Resolution Slide-seqV2 Spatial Transcriptomics Enables Discovery of Disease-Specific Cell Neighborhoods and Pathways
> https://cellxgene.cziscience.com/collections/8e880741-bf9a-4c8e-9227-934204631d2a
>
> High resolution spatial transcriptomics is a transformative technology that enables mapping of RNA expression directly from intact tissue sections; however, its utility for the elucidation of disease processes and therapeutically actionable pathways remain largely unexplored. Here we applied Slide-seqV2 to mouse and human kidneys, in healthy and in distinct disease paradigms. First, we established the feasibility of Slide-seqV2 in human kidney by analyzing tissue from 9 distinct donors, which revealed a cell neighborhood centered around a population of LYVE1+ macrophages. Second, in a mouse model of diabetic kidney disease, we detected changes in the cellular organization of the spatially-restricted kidney filter and blood flow regulating apparatus. Third, in a mouse model of a toxic proteinopathy, we identified previously unknown, disease-specific cell neighborhoods centered around macrophages. In a spatially-restricted subpopulation of epithelial cells, we also found perturbations in 77 genes associated with the unfolded protein response (UPR), including Tmed9. Treatment with a TMED9-targeting compound showed efficient removal of toxic mutant proteins and reversal of the UPR. Our studies illustrate and experimentally validate the utility of Slide-seqV2 for the discovery of disease-specific cell neighborhoods and actionable targets.
>
> dataset_h5ad_path:
 
> ### Spatiotemporal immune zonation of the human kidney
> https://cellxgene.cziscience.com/collections/120e86b4-1195-48c5-845b-b98054105eec
> 
> Single-cell RNA sequencing has begun to shed light on the full cellular diversity of specific organs. However, these studies rarely examine organ-specific immune cells. Stewart et al. sequenced healthy adult and fetal kidney samples at a single-cell level to define the heterogeneity in epithelial, myeloid, and lymphoid cells. From this dataset, they identified zonation of cells, with relevance to disease and the varied perturbations that occur in different tumor settings. This profiling of the human kidney generates a comprehensive census of existing cell populations that will help inform the diagnosis and treatment of kidney-related diseases.
> 
> dataset_h5ad_path: 

> ### A single-cell survey of the human first-trimester placenta and decidua (10x)
> https://www.ebi.ac.uk/gxa/sc/experiments/E-HCAD-24/results/cell-plots

> ### A human cell atlas of fetal gene expression
> https://cellxgene.cziscience.com/collections/c114c20f-1ef4-49a5-9c2e-d965787fb90c
> 
> The gene expression program underlying the specification of human cell types is of fundamental interest. We generated human cell atlases of gene expression and chromatin accessibility in fetal tissues. For gene expression, we applied three-level combinatorial indexing to >110 samples representing 15 organs, ultimately profiling ~4 million single cells. We leveraged the literature and other atlases to identify and annotate hundreds of cell types and subtypes, both within and across tissues. Our analyses focused on organ-specific specializations of broadly distributed cell types (such as blood, endothelial, and epithelial), sites of fetal erythropoiesis (which notably included the adrenal gland), and integration with mouse developmental atlases (such as conserved specification of blood cells). These data represent a rich resource for the exploration of in vivo human gene expression in diverse tissues and cell types.

> ### Human Brain Cell Atlas v1.0
> https://cellxgene.cziscience.com/collections/283d65eb-dd53-496d-adb7-7570c7caa443
> 
> First draft atlas of human brain transcriptomic cell types: The human brain directs a wide range of complex behaviors ranging from fine motor skills to abstract intelligence and emotion. This broad range of functions is supported by an exceptionally complex cellular and circuit architecture. To create a first draft human brain cell atlas, high-throughput single-nucleus RNA sequencing was used to systematically survey cells across the entire adult human brain in three postmortem donors. Over three million nuclei were sampled from approximately 100 dissections across the forebrain, midbrain, and hindbrain. Analysis of these data showed regional diversity in that cellular organization exhibited regional diversity at multiple scales, identifying 30 superclusters, 461 clusters and 3313 subclusters. As the first single-cell transcriptomic census of the entire human brain, this atlas provides a resource for understanding the molecular diversity of the human brain in health and disease. The Human Brain Cell Atlas v1.0 is presented for visualization and data mining through the Chan Zuckerberg Initiative’s CellxGene application, with the following biologically meaningful partitions: 1. Neuronal and non-neuronal cell types 2. Supercluster-specific groupings (”Supercluster: ”) 3.Brain region-specific groupings (”Dissection: “), ordered by the adult human brain anatomical reference atlas ontology in Ding et al. (2016)
> 
> Used 'All non-neuronal cells' for the Brain data

> ### Single-Cell DNA Methylation and 3D Genome Human Brain Atlas
> https://cellxgene.cziscience.com/collections/fdebfda9-bb9a-4b4b-97e5-651097ea07b0
> 
> Description: Delineating the gene regulatory programs underlying complex cell types is fundamental for understanding brain functions in health and disease. Here, we comprehensively examine human brain cell epigenomes by probing DNA methylation and chromatin conformation at single-cell resolution in over 500,000 cells from 46 brain regions. We identified 188 cell types and characterized their molecular signatures. Integrative analyses revealed concordant changes in DNA methylation, chromatin accessibility, chromatin organization, and gene expression across cell types, cortical areas, and basal ganglia structures. With these resources, we developed scMCodes that reliably predict brain cell types using their methylation status at select genomic sites. This multimodal epigenomic brain cell atlas provides new insights into the complexity of cell type-specific gene regulation in the adult human brain


## Mice Database

> ### A comprehensive mouse kidney atlas enables rare cell population characterization and robust marker discovery
> https://cellxgene.cziscience.com/collections/92fde064-2fb4-41f8-b85c-c6904000b859
> 
> Used to get Renal System data for _Mus musculus_
> 
> dataset_h5ad_path: 42bb7f78-cef8-4b0d-9bba-50037d64d8c1.h5ad

> ### Tabula Muris Senis: The Aging Mouse Transcriptome Database
> https://cellxgene.cziscience.com/collections/0b9d8a04-bb9d-44da-aa27-705bb65b54eb
> 
> Used for Kidney and Brain data

> ### DNA Methylation Atlas of the Mouse Brain at Single-Cell Resolution
> https://cellxgene.cziscience.com/collections/37f1f46d-6dfa-482c-9b17-d5850d8536f6
> 
> Mammalian brain cells are remarkably diverse in gene expression, anatomy, and function, yet the regulatory DNA landscape underlying this extensive heterogeneity is poorly understood. We carried out a comprehensive assessment of the epigenomes of mouse brain cell types by applying single nucleus DNA methylation sequencing to profile 110,294 nuclei from 45 regions of the mouse cortex, hippocampus, striatum, pallidum, and olfactory areas. We identified 161 cell clusters with distinct spatial locations and projection targets. We constructed taxonomies of these epigenetic types, annotated with signature genes, regulatory elements, and transcription factors. These features indicate the potential regulatory landscape supporting the assignment of putative cell types, and reveal repetitive usage of regulators in excitatory and inhibitory cells for determining subtypes. The DNA methylation landscape of excitatory neurons in the cortex and hippocampus varied continuously along spatial gradients. Using this deep dataset, an artificial neural network model was constructed that precisely predicts single neuron cell-type identity and brain area spatial location. Integration of high-resolution DNA methylomes with single-nucleus chromatin accessibility data allowed prediction of high-confidence enhancer-gene interactions for all identified cell types, which were subsequently validated by cell-type-specific chromatin conformation capture experiments. By combining multi-omic datasets (DNA methylation, chromatin contacts, and open chromatin) from single nuclei and annotating the regulatory genome of hundreds of cell types in the mouse brain, our DNA methylation atlas establishes the epigenetic basis for neuronal diversity and spatial organization throughout the mouse brain.

# Miscellaneous

> ### Single-cell Best Practices
> https://www.sc-best-practices.org/preprocessing_visualization/quality_control.html

> ### Cell Hint
> https://cellhint.readthedocs.io/en/latest/notebook/cellhint_tutorial_integration.html
> 
> CellHint is an automated tool for cell type harmonisation and integration
> - harmonisation: match and harmonise cell types defined by independent datasets
> - integration: integrate cell and cell types with supervision from harmonisation

> 
